### PR TITLE
[Vaisala] Quality Reusable Resource - 6042

### PR DESCRIPTION
**Pull request by seanmcilroy29**
See Issue #151

Vaisala submission for new Reusable Resource

### CODE REVIEW

**Comment by akeranen** [comment]
@dirtyha would it be possible to have standardized (for this resource) values for quality of measurement? For example a range with semantics or a set of enumerations? That would make this resource much more useful since it could interoperate across systems. Otherwise perhaps we should call this "Vendor specific quality" or similar? 

Also in the Porto meeting we decided to use two dots ("..") as the range indicator so the range should be "-8999..8999". Unfortunately there's no public documentation of this convention yet.

Would be interesting to learn about the reason for this specific range too.

File: Common.xml [hunk 1]
```diff
@@ -3435,13 +3435,13 @@ Using values in range 0-127 is recommended to avoid ambiguities with byte order
...
       
         <Item ID="6042">    
-        <Name>Reserved</Name>     
-        <Operations></Operations>     
-        <Type></Type>
-        <RangeEnumeration></RangeEnumeration>
+        <Name>Quality</Name>     
+        <Operations>R</Operations>     
+        <Type>Integer</Type>
+        <RangeEnumeration>-8999 to 8999</RangeEnumeration>
```
Comment (mcocak) [inline Common.xml]: Is there a reference for this range? Or is it some proprietary range?

**Comment by seanmcilroy29** [comment]
Comment submitted to Issue #151 by [@dirtyha](https://github.com/dirtyha) 

Due to interest on this specific quality value range, we decided to open it a little bit. Basically it is a Vaisala specific coding used by our existing products. If a measurement device has quality control algorithms, it may then report:

INVALID (-8500) : Measurement value shall not be used
UNCHECKED (0) : Measurement value has not undergone any quality checks
SUSPICIOUS (6500) : Measurement value contains uncertainty, for example statistic with low number of samples
ACCEPTED (8500): Measurement value is good for use

Other values in the range (-8999..8999) are reserved for quality levels that are done at later phases at server side. Generally speaking, the larger the value, the "better" the measurement is.

File: Common.xml [hunk 2]
```diff
@@ -3434,14 +3434,14 @@ Using values in range 0-127 is recommended to avoid ambiguities with byte order
         <TSLink>0</TSLink>
       </Item>
       
-        <Item ID="6042">    
-        <Name>Reserved</Name>     
-        <Operations></Operations>     
-        <Type></Type>
-        <RangeEnumeration></RangeEnumeration>
+      <Item ID="6042">    
+        <Name>Quality</Name>
```
Comment (mkgillmore) [inline Common.xml:3438]: Consider changing name to Vaisala Quality

File: Common.xml [hunk 3]
```diff
@@ -3434,14 +3434,14 @@ Using values in range 0-127 is recommended to avoid ambiguities with byte order
...
+      <Item ID="6042">    
+        <Name>Quality</Name>     
+        <Operations>R</Operations>     
+        <Type>Integer</Type>
+        <RangeEnumeration>-8999..8999</RangeEnumeration>
         <Units></Units>
-        <Submitter></Submitter>
-        <Description></Description>
+        <Submitter>Vaisala</Submitter>
+        <Description>Quality value indicates how reliable sensor measurement is. This coding is vendor specific.</Description>
```
Comment (mkgillmore) [inline Common.xml:3444]: IPSO Group agrees to potentially merge the object with the following added to the description:
INVALID (-8500) : Measurement value shall not be used
UNCHECKED (0) : Measurement value has not undergone any quality checks
SUSPICIOUS (6500) : Measurement value contains uncertainty, for example statistic with low number of samples
ACCEPTED (8500): Measurement value is good for use

Remove The sentence "This coding is vendor specific"

Consider defining what the values between (e.g. -8500 thru 0) represent in the description.

**Comment by dirtyha** [comment]
Vaisala is ready to make this re-usable resource more generic with a simple range and rename it "Measurement Quality". After a call with @akeranen , we concluded that for clarity sake, it might be better to close this PR as rejected and create a new registration for the same resource id 6042.

**Comment by jpradocueva** [comment]
@dirtyha  As suggested this PR is closed. We will closed as well the branch "Vaisala" and we will create a new  new branch "Vaisala-1" once it is done it will be notified.

**Comment by jpradocueva** [comment]
@dirtyha a new branch called "Vaisala-1" has been created.
Please apply your PR against the Common.xml document Reusable Resource 6042:
https://github.com/OpenMobileAlliance/lwm2m-registry/blob/Vaisala-1/Common.xml

You have been added to "lwm2m-registry" as a collaborator with READ access writes. You should see the invitation here: https://github.com/OpenMobileAlliance/lwm2m-registry/invitations

Let us know if you need any assistance.

**Pull request closed without merging**